x/auth/ante/authz_rules_ante.go

```diff
@@ -0,0 +1,123 @@
+package ante
+
+import (
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authztypes "github.com/cosmos/cosmos-sdk/x/authz"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+const (
+	AllowedRecipients = "allowed_recipients"
+	MaxAmount         = "max_amount"
+)
+
+type AuthzDecorator struct {
+	azk AuthzKeeper
+	ak  AccountKeeper
+}
+
+func NewAuthzDecorator(azk AuthzKeeper, ak AccountKeeper) AuthzDecorator {
+	return AuthzDecorator{
+		azk: azk,
+		ak:  ak,
+	}
+}
+
+// AuthzDecorator checks the authorization message grants for some rules.
+func (azd AuthzDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
+	if !ok {
+		return ctx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "invalid tx type")
+	}
+
+	signers, err := sigTx.GetSigners()
+	if err != nil {
+		return ctx, err
+	}
+
+	grantee := signers[0]
+
+	msgs := tx.GetMsgs()
+	for _, msg := range msgs {
+		// Check if the message is an authorization message
+		if authzMsg, ok := msg.(*authztypes.MsgExec); ok {
+
+			msgs, err := authzMsg.GetMessages()
+			if err != nil {
+				return ctx, err
+			}
+
+			for _, innerMsg := range msgs {
+				switch innerMsgConverted := innerMsg.(type) {
+				case *banktypes.MsgSend:
+					isRulesBroken, err := azd.handleSendAuthzRules(ctx, innerMsgConverted, grantee)
```

Comment:
If the rules are broken, it should err.

```suggestion
					if err := azd.handleSendAuthzRules(ctx, innerMsgConverted, grantee); err != nil) {
					return err
					}
```